### ISSUE TITLE
Problem with width if I use chosen on a modal form

### ISSUE THREAD

**Issue by didebeach** [leocaseiro/angular-chosen#81]
I'm very happy with the directive. Current I do have a small problem if I try to use it on a modal form (with bootstrap) as the width is always 0.

I'm able to reproduce it http://plnkr.co/edit/NPdH9wBrXBLiAjE5yMlN?p=preview and on the example I show on the first screen a "regular" chosen that is working and if you press the button you will see that it's not working the same.

I did have a look inside of the chosen.jquery.js file and I found that on the line 433 may I have the possibility to handover some options but I could not find out how I can use / set the options. I always fall into the else case and the >>this.form_field.offsetWidth << is always 0 in this case.

Any idea would be really great

Dirk

**Comment by lpsBetty** [comment]
I noticed that the chosen element gets the width of the select .. so e.g. if you have a select with the bootstrap class 'form-control' the width of the chosen will always be 100%..

but if you want your chosen to be exactly the width of the content you can do this:

````
.chosen-container{
    width: auto !important;
}
````
or you give the container a min-width..

**Comment by bitakazemi** [comment]
<script>
$("select:not('.chosen-specify-width')").chosen({width: '90%'});
$(".chosen-specify-width").chosen({width: '90%'}).show();
$("div").show();

</script>
<style>
adding this script helped me, I found it here :
http://jsfiddle.net/geJ7t/1/

**Comment by bitakazemi** [comment]
I tried a couple of solutions but using this script on top of my html was the only thing that worked for me.
```javascript
$("select:not('.chosen-specify-width')").chosen({width: '90%'});
$(".chosen-specify-width").chosen({width: '90%'}).show();
$("div").show();
```




http://jsfiddle.net/geJ7t/1/

**Comment by devilankur18** [comment]
This worked for me 


    .chosen-container{
      width: 100% !important;
    }

**Comment by leocaseiro** [comment]
To fix this issue which is duplicate of #42 you can set the ` width="'100%'"` and it'll fix your issue:

Your plunkr with the solution http://plnkr.co/edit/XcQzmH?p=preview
